app-creator/deserializer.js

```diff
@@ -0,0 +1,438 @@
+    case 'checkbox':
+      return app.createCheckboxElement(nodeObj, filteredStyles);
+    case 'textbox':
+      return app.createTextboxElement(nodeObj, filteredStyles);
+    case 'select':
+      return app.createSelectElement(nodeObj, filteredStyles);
+    case 'slider':
+      return app.createSliderElement(nodeObj, filteredStyles);
+    case 'chart':
+      return app.createChartElement(nodeObj, filteredStyles);
+    default:
+      return { node: null };
+  }
+};
+
+/**
+ * Helper function for creating slider elements.
+ */
+app.createSliderElement = function (obj, style) {
+  var uniqueAttributes = obj.uniqueAttributes;
+
+  var min = parseFloat(uniqueAttributes.min);
+  min = isNaN(min) ? 0 : min;
+
+  var max = parseFloat(uniqueAttributes.max);
+  max = isNaN(max) ? 1 : max;
+
+  var step = parseFloat(uniqueAttributes.step);
+  step = isNaN(step) ? 0.01 : step;
+
+  var value = parseFloat(uniqueAttributes.value);
+  value = isNaN(value) ? 0 : value;
+
+  var direction = uniqueAttributes.direction;
+
+  var disabled = uniqueAttributes.disabled === 'true';
+
+  var slider = ui.Slider({
+    min: min,
+    max: max,
+    value: value,
+    step: step,
+    direction: direction,
+    disabled: disabled,
+    style: style,
+  });
+
+  return { node: slider };
+};
+
+/**
+ * Helper function for creating chart elements.
+ */
+app.createChartElement = function createChartElement(obj, style) {
+  var uniqueAttributes = obj.uniqueAttributes;
+
+  var dataTable = uniqueAttributes.dataTable;
+
+  var chartType = uniqueAttributes.chartType.replace(/\s/g, '');
+
+  var title = uniqueAttributes.title;
+
+  var color = uniqueAttributes.color;
+  color = color.split(',');
+  for (var i = 0; i < color.length; i++) {
+    color[i] = color[i].trim();
+  }
+
+  var threeD = uniqueAttributes['3D'];
+
+  var downloadable = uniqueAttributes.downloadable === 'true';
+
+  var chart = ui.Chart({
+    dataTable: {
+      cols: [
+        { id: 'task', label: 'Task', type: 'string' },
+        { id: 'hours', label: 'Hours per Day', type: 'number' },
+      ],
+      rows: [
+        { c: [{ v: 'Eat' }, { v: 2 }] },
+        { c: [{ v: 'Write EE Code' }, { v: 9 }] },
+        { c: [{ v: 'Sleep' }, { v: 7, f: '7.000' }] },
+      ],
+    },
+    options: {
+      title: title,
+      color: color,
+      is3D: threeD,
+      height: style.height,
+      width: style.width,
+    },
+    chartType: chartType,
+    downloadable: downloadable,
+  });
+
+  return { node: chart };
+};
+
+/**
+ * Helper function for creating select elements.
+ */
+app.createSelectElement = function createSelectElement(obj, style) {
+  var uniqueAttributes = obj.uniqueAttributes;
+
+  var items = uniqueAttributes.items;
+  items = items.split(',');
+  for (var i = 0; i < items.length; i++) {
+    items[i] = items[i].trim();
+  }
+
+  var placeholder = uniqueAttributes.placeholder;
+
+  var value = items.length > 0 ? items[0] : '';
+
+  var disabled = uniqueAttributes.disabled === 'true';
```

Comment:
if we serialize it into a boolean and call JSON.parse like so:

```Javascript
var value = JSON.parse('{"shown":true}');
```
The type of value.shadow is JSON. 

If we want to retrieve the boolean, we will have to call JSON.parse on the boolean itself like so: 
```javascript
JSON.parse(value.shown)
```

We can go with either approach. But I think since we don't get the value right away and have to do the extra parsing step, we might as well compare the string value of the boolean. Thoughts?

Reply:
That's interesting. I just ran it in the console and it seemed to work as expected:
```js
JSON.parse('{"shown":true}').shown
```
Outputs: `true`

Perhaps there's a bug somewhere which is adding an extra object wrapper?